src/include/platform/internal/GenericConfigurationManagerImpl.ipp

```diff
@@ -873,6 +902,20 @@ void GenericConfigurationManagerImpl<ConfigClass>::LogDeviceConfig()
     }
 }
 
+template <class ConfigClass>
+void GenericConfigurationManagerImpl<ConfigClass>::HandleFailSafeContextCleanup(intptr_t arg)
+{
+    if (ConfigurationMgr().SetFailSafeArmed(false) != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Failed to set FailSafeArmed config to false");
+    }
+
+    if (FailSafeContext::DeleteFromStorage() != CHIP_NO_ERROR)
```

Comment:
So I think there is one more race here.  Say we boot up and detect we had a failsafe that has not been cleaned up.  We do this async cleanup, but we are _not_ setting the fail-safe context to be armed in the meantime.  So an ArmFailSafe command can come in and overwrite the storage (and our stored "fail-safe is armed") boolean, then this fires and removes the storage, and then if we reboot while the new fail-safe is live it will never get cleaned up.

We really need to figure out an actually consistent way to keep track of our state.  My suggestion would be:

1. When fail-safe is set up, it sets its boolean to armed and stores the boolean in the configuration manager (which this PR does).
2. When mutations to the fail-safe happen those get stored (which this PR does).
3. When the fail-safe time fires, the fail-safe is placed into a mode where it looks like it's no longer armed for purposes of most commands that depend on a fail-safe but still blocks arming a new fail-safe (this we do not have right now) and stays in that mode while the async cleanups happen.  Once those complete it's marked fully disarmed.  Probably need a second boolean in the fail-safe context.
4. On startup, if we discover that the fail-safe was not cleaned up then we put it in the same mode as in item 3, then queue the async cleanups.  Ideally by calling a method on the fail-safe so all that code is shared....

This should ensure that once the timer expires, or we shut down, things that need a fail-safe to function error out, but we don't allow creation of a new fail-safe and overwriting our storage until we have ensured cleanup is complete.